app/views/spree/products/_cart_form.html.erb

```diff
@@ -85,6 +85,9 @@
 <% default_variant = default_variant(@variants, @product) %>
 <div class="product-side-details d-flex flex-column justify-content-between h-100"  >
   <div class="top-detail" >
+    <% if @product.on_sale? %>
+      <div class="deal-expiry-wrap" id="demo" data-SalesExpiryDate="<%= @product.master.sale_prices.active.last.end_at.strftime('%b %d, %Y %H:%M:%S')%>"></div>
```

Comment:
create instance method or helper of this
master.sale_prices.active.last.end_at.strftime('%b %d, %Y %H:%M:%S')